akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/EventSourcedBehavior.scala

```diff
@@ -140,16 +139,28 @@ abstract class EventSourcedBehavior[Command, Event, State] private[akka] (
    * When persisting multiple events at once the snapshot is triggered after all the events have
    * been persisted.
    *
-   * Snapshots triggered by `snapshotWhen` will not trigger deletes of old snapshots and events if
+   * Snapshots triggered by `shouldSnapshot` will not trigger deletes of old snapshots and events if
    * [[EventSourcedBehavior.retentionCriteria]] with [[RetentionCriteria.snapshotEvery]] is used together with
    * `shouldSnapshot`. Such deletes are only triggered by snapshots matching the `numberOfEvents` in the
    * [[RetentionCriteria]].
    *
+   * Events can be deleted if `deleteEvnetsOnShouldSnapshot` returns `true`.
+   *
    * @return `true` if snapshot should be saved at the given `state`, `event` and `sequenceNr` when the event has
    *         been successfully persisted
    */
   def shouldSnapshot(@unused state: State, @unused event: Event, @unused sequenceNr: Long): Boolean = false
 
+  /**
+   * Can be used to delete events after `shouldSnapshot`.
+   *
+   * Can be used in combination with `[[EventSourcedBehavior.retentionCriteria]]` in a way that events are triggered
+   * up the the oldest snapshot based on `[[RetentionCriteria.snapshotEvery]]` config.
+   *
+   * @return `true` if events should be deleted
+   */
+  def deleteEventsOnShouldSnapshot: Boolean = false
```

Comment:
This name reads rather funny, but I see how it is logically correct given the shouldSnapshot name.
Maybe it would be enough with `deleteEventsOnSnapshot`, or do you think that is ambiguous with the retention criteria?

Reply:
`deleteEventsOnSnapshot` seems the better choice 👍